src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ListViewBase.cs

```diff
@@ -637,6 +646,71 @@ public async Task NoItemSelectedSingle()
 			Assert.AreEqual(list.SelectedIndex, -1);
 		}
 
+		[TestMethod]
+		[RunsOnUIThread]
+		public async Task When_Multiple_Selection_Pointer()
+		{
+			var items = Enumerable.Range(0, 10).Select(i => new ListViewItem { Content = i}).ToArray();
+			var list = new ListView
+			{
+				SelectionMode = ListViewSelectionMode.Multiple,
+			};
+
+			items.ForEach((ListViewItem item) => list.Items.Add(item));
+
+			WindowHelper.WindowContent = list;
+			await WindowHelper.WaitForIdle();
+
+			var injector = InputInjector.TryCreate() ?? throw new InvalidOperationException("Failed to init the InputInjector");
+			using var mouse = injector.GetMouse();
+
+			var selected = new List<ListViewItem>();
+			await AssertSelected();
+
+			mouse.Press(Center(items[1]));
+			mouse.Release();
+
+			selected.Add(items[1]);
+			await AssertSelected();
+
+			mouse.Press(Center(items[3]), VirtualKeyModifiers.Shift);
+			mouse.Release(VirtualKeyModifiers.Shift);
+
+			selected.AddRange(items.Where((_, i) => i is > 1 and <= 3).ToList());
+			await AssertSelected();
+
+			mouse.Press(Center(items[6]));
+			mouse.Release();
+
+			selected.Add(items[6]);
+			await AssertSelected();
+
+			mouse.Press(Center(items[8]), VirtualKeyModifiers.Shift);
+			mouse.Release(VirtualKeyModifiers.Shift);
+
+			selected.AddRange(items.Where((_, i) => i is > 6 and <= 8));
+			await AssertSelected();
+
+			mouse.Press(Center(items[8]));
+			mouse.Release();
+
+			selected.Remove(items[8]);
+			await AssertSelected();
+
+			mouse.Press(Center(items[4]), VirtualKeyModifiers.Shift);
```

Comment:
For some reason. this refuses to work. `mouse.Press(Center(items[4]), VirtualKeyModifiers.Shift);` actually clicks on the first item (index 0). `mouse.Press(Center(items[0]), VirtualKeyModifiers.Shift);` doesn't click on anything. I don't understand why this only happens here and not in the calls above.

Reply:
The problem was that the interpolation algorithm used to calculate the pointer movement steps in InputInjector was very incorrect.